view/vocabularylist.twig

```diff
@@ -15,7 +15,14 @@
       <h3>{{ vocabClassName }}</h3>
       <ul>
       {% for vocab in vocabArray %}
-        <li><a class="navigation-font" href="{{ vocab.id }}/{{ request.lang }}/{% if request.contentLang != request.lang and request.contentLang != '' and request.contentLang in vocab.config.languages %}?clang={{ request.contentLang }}{% endif %}">{{ vocab.title }}</a></li>
+        {% set useDefaultLanguage = null %}
+        {% if request.contentLang not in vocab.config.languages and request.contentLang != '' %}
+          {% set useDefaultLanguage = true %}
+        {% endif %}
+        <li><a class="navigation-font" href="{{ vocab.id }}/{{ request.lang }}/{% if request.contentLang not in vocab.config.languages and request.contentLang != '' %}?clang={{ vocab.config.defaultLanguage }}{% endif %}">
+          {% if useDefaultLanguage is not null %} {{ vocab.title(vocab.config.defaultLanguage) }}
+            {% else %} {{ vocab.title(request.contentLang) }}
+          {% endif %} </a></li>
```

Comment:
Couldn't we use something like (not tested) this?

```suggestion
        {% set useDefaultLanguage = request.contentLang not in vocab.config.languages and request.contentLang != '' %}
        <li><a class="navigation-font" href="{{ vocab.id }}/{{ request.lang }}/{% if useDefaultLanguage %}?clang={{ vocab.config.defaultLanguage }}{% endif %}">
          {% if useDefaultLanguage %} {{ vocab.title(vocab.config.defaultLanguage) }}
          {% else %} {{ vocab.title(request.contentLang) }}
          {% endif %} </a></li>
```
I think we have the same condition in two places, so maybe storing the `true` or `false` value of that condition in that variable would simplify it?

Reply:
The way you proposed was clearly more elegant, I changed the code. Thanks a lot @kinow